NDTensors/src/abstractarray/ndims.jl

```diff
@@ -0,0 +1,7 @@
+ndims(x) = Base.ndims(x)
+
+if VERSION < v"1.7"
+  ndims(x::AbstractArray) = ndims(typeof(x))
```

Comment:
Maybe better to use:
```julia
ndims(array::AbstractArray) = Base.ndims(array)
```
so that it explicitly uses the `Base` Julia definition.

Also, let's add a comment linking to https://github.com/JuliaLang/julia/pull/40682 so we remember why we had to do this.